src/core/namespace.c

```diff
@@ -1019,6 +1028,30 @@ static int mount_images(const MountEntry *m) {
         return 1;
 }
 
+static int mount_overlay(const MountEntry *m) {
+        _cleanup_free_ char *escaped_source = NULL;
+        const char *options, *where, *prefixed_source;
+
+        assert(m);
+
+        where = prefix_roota(mount_entry_path(m), m->overlay_malloc);
+
+        (void) mkdir_p_label(where, 0755);
+
+        prefixed_source = prefix_roota(mount_entry_source(m), m->overlay_malloc);
+        escaped_source = shell_escape(prefixed_source, ",:");
+        if (!escaped_source)
+                return -ENOMEM;
+
+        /* We only support read-only overlays, so no upper nor work directories */
+        options = strjoina("lowerdir=", escaped_source, ":", where);
```

Comment:
`where` needs to be escaped too, no?

Reply:
yes indeed, it goes as an option too, good catch